README.md

```diff
@@ -771,28 +777,85 @@ let numbers = create([Number], [1, 2, 3, 4]);
 
-Microstates provides an easy way to convert a Microstate which represents a single value into
-a Observable stream of values. This is done by passing a Microstate to `Observable.from` function.
-This function will return a Observable object with a `subscribe` method. You can subscribe to the
-stream by passing an observer to the subscribe function. Once you subscribe, you will synchronously
-receive a microstate with middleware installed that will cause the result of transitions to be pushed
-through the stream.
+To bring a reactive system to life, we need to capture the latest state in a sequence of states. In component driven reactive engines like React, Vue, Ember & Angular the state needs to emerge at the root of the component tree. When a new state is emitted, we need to apply the state to the root component and allow the state to cascade down the component tree.
 
-You should be able to use to any implementation of Observables that supports `Observer.from` using [symbol-observable](https://github.com/benlesh/symbol-observable). We'll use `RxJS` for our example.
+Microstates has a mechanism to working with Microtates in reactive environments. It is called Identity. When you create an Identity from a Microstate, you get an object that has the same shape the original microstate. Every composed microstate becomes an identity and every transition gets wrapped in side effect emitting behaviour specific to that identity constructor. Let’s look at the features of the Identity and how they benefit reactive environments.
+
+## Structural Sharing
+
+A common performance optimization used by all reactive engines is to prevent re-renders for components who’s props have not changed. The most efficient way to determine if a value has not changed it to perform an exact equality check, for example: `prevValue === currentValue`. If the reference is the same, then consider the value unchanged. The Identity makes this possible with Microstates by internally managing how the Identity is constructed as a result of a transition. It will automatically determine which branches of microstates are unchanged and reuse previous identities for those branches.
+
+## Memoized Getters
+
+Microstates are immutable which makes it safe for us to memoize computations that are derived off their state. Identies will automatically memoize getters and return previously computed value when the microstate backing the identity has not changed. When a transition is invoked on the identity, the part of the identity tree that are changed will be re-created effectively invalidating the cache for the changed parts of the identity. The getters will be recomputed for state that is changed.
+
+## Debounce no-op transitions
```

Comment:
How is a no-op calculated? By which I mean to ask, a transition from and back to the original state within what time period avoids emitting a new Identity? The following code performs two transitions

```javascript
let id = Store(from({ foo: 'bar' }), prev => {
  console.count('changed');
  id = prev;
});

id.foo.set('bax');
id.foo.set('bar');
```

Reply:
> How is a no-op calculated?

It's based on value equality. Using your example,

```js
id.foo.set('bax');
id.foo.set('bax');
```

Would only cause one update because 2nd operation is debounced.

Reply:
Oh okay, I see what you're saying -- if the current state and the state being transitioned to are the same, then a "new state" that would be the same _is not_ emitted.

Reply:
Yes.